api/models/trips.models.js

```diff
@@ -58,3 +59,61 @@ exports.postTrip = (newTrip) => {
       return tripEntered;
     });
 };
+
+exports.doesTripExist = (trip_id) => {
+  if (Number.isInteger(+trip_id) || trip_id.length < 24) {
+    return Promise.reject({
+      status: 400,
+      msg: `trip_id '${trip_id}' is an invalid trip ID.`,
+    });
+  } else {
+    return trips.findOne({ _id: new ObjectId(trip_id) }).then((trip) => {
+      if (trip === null) {
+        return Promise.reject({
+          status: 404,
+          msg: `trip_id '${trip_id}' does not exist.`,
+        });
+      }
+    });
+  }
+};
+
+exports.selectSingleTrip = (trip_id, username) => {
+  return Promise.all([selectUsername(username), this.doesTripExist(trip_id)])
+    .then(() => {
+      const query = {
+        _id: new ObjectId(trip_id),
+        attending: { $in: [username] },
+      };
+      return trips.findOne(query);
+    })
+    .then((trip) => {
+      if (trip === null) {
+        return Promise.reject({
+          status: 401,
+          msg: "You are unauthorised to access this trip.",
+        });
+      } else {
+        return trip;
+      }
+    });
+};
+
+exports.removeTrip = (trip_id, username) => {
+  return Promise.all([selectUsername(username), this.doesTripExist(trip_id)])
+    .then(() => {
+      const query = {
+        _id: new ObjectId(trip_id),
+        "attending.0": username,
```

Comment:
We could consider adding a 'creator' field in the future/if we get time before launch, or allowing any user in attending to delete the trip

The user at 0th position will normally be the creator, but if that user leaves a multi-user trip then the 'creator' is seen as the next user (the one that was in the 1st position).

Reply:
I think having a creator field would definitely be a good idea! It would be cool if only the creator could delete the trip. I'll add it to the "future" list!

On the PATCH, I am going to ensure that the creator status can be passed to someone else who is attending (i.e. the new creator is moved to the start of the array) - hopefully this should solve it for now - we can chat about it in stand up! :)